components/homme/src/theta-l_kokkos/prim_driver_mod.F90

```diff
@@ -93,29 +93,39 @@ subroutine prim_create_c_data_structures (tl, hvcoord, mp)
     type (c_ptr) :: hybrid_am_ptr, hybrid_ai_ptr, hybrid_bm_ptr, hybrid_bi_ptr
     character(len=MAX_STRING_LEN), target :: test_name
 
+    integer :: disable_diagnostics_int, theta_hydrostatic_mode_int, use_moisture_int
+
     ! Initialize the C++ reference element structure (i.e., pseudo-spectral deriv matrix and ref element mass matrix)
     dvv = deriv1%dvv
     elem_mp = mp
     call init_reference_element_c(c_loc(dvv),c_loc(elem_mp))
 
     ! Fill the simulation params structures in C++
     test_name = TRIM(test_case) // C_NULL_CHAR
+
```

Comment:
This seems overly complicated. We create an int from a bool to call the c function, and then inside the c function we convert the int back to a bool. Why not passing bool in the first place?